mtgjson5/providers/tcgplayer.py

```diff
@@ -148,6 +148,71 @@ def generate_today_price_dict(
         return dict(results)
 
 
+def get_tcgplayer_sku_data(group_id_and_name: Tuple[str, str]) -> List[Dict]:
+    """
+    finds all sku data for a given group using the TCGPlayer API
+    :param group_id_and_name: group id and name for the set to get data for
+    :return: product data including skus to be parsed into a sku map
+    """
+    magic_set_product_data = []
+    api_offset = 0
+
+    while True:
+        api_response = TCGPlayerProvider().download(
+            "https://api.tcgplayer.com/catalog/products",
+            {
+                "offset": str(api_offset),
+                "limit": 100,
+                "categoryId": 1,
+                "includeSkus": True,
+                "groupId": group_id_and_name[0],
+            },
+        )
+
+        if not api_response:
+            # No more entries
+            break
+
+        response = json.loads(api_response)
+        if not response["results"]:
+            # Something went wrong
+            break
+
+        for product_data in response["results"]:
+            magic_set_product_data.append(product_data)
+
+        api_offset += len(response["results"])
+
+    return magic_set_product_data
+
+
+def generate_tcgplayer_sku_map(tcgplayer_set_sku_data: List[Dict]) -> Dict[str, List]:
+    """
+    takes product info and builds a sku map
+    :param tcgplayer_set_sku_data: list of product data dicts used to a build a product id to sku map
+    :return: Map of TCGPlayerID -> NM Foil and Nonfoil SKU
+    """
+    tcgplayer_sku_map: Dict[str, List] = {}
+    for product_data in tcgplayer_set_sku_data:
+        nonfoil_sku: int = 0
+        foil_sku: int = 0
+        for sku in product_data["skus"]:
+            if (
+                sku["conditionId"] == 1
+                and sku["printingId"] == 1
+                and sku["languageId"] == 1
+            ):
+                nonfoil_sku = sku["skuId"]
+            elif (
+                sku["conditionId"] == 1
+                and sku["printingId"] == 2
+                and sku["languageId"] == 1
+            ):
+                foil_sku = sku["skuId"]
+        tcgplayer_sku_map[str(product_data["productId"])] = [nonfoil_sku, foil_sku]
```

Comment:
If you're returning a specific ordering, I'd consider making it a dict instead. {"foil": ..., "nonfoil": ...} for example.